### DISCUSSION TITLE
Info regarding Dropbox sync support - reached preliminary limit of 50 accounts

### DISCUSSION THREAD

**Discussion by tobias-klein** [ezra-bible-app/ezra-bible-app#1082] [General]
The amount of people using the Dropbox sync feature has been steadily growing. The number of linked accounts (measured through Dropbox developer console) has recently reached fifty and that is a limit defined by Dropbox. Dropbox has now asked me to apply for *production* mode which would allow more users. If you are a new user and you are having trouble linking Ezra Bible App to your Dropbox, this limit may be the reason. There are a few things I need to do technically to be qualified for Dropbox production access. One of the things that needs to be addressed is #1079. I am going to try to make it happen for the next release. However, it depends on the approval by the Dropbox folks. So, in the meantime Dropbox sync will continue working for existing users. New users may not be able to connect anymore - due to the user limit that we have reached.

**Comment by tobias-klein** [comment]
I made some changes in release 1.14.1 - to address #1079. Based on this and some changes on the requested Dropbox permissions, I will re-apply for production mode and see what the Dropbox folks say.

**Comment by tobias-klein** [comment]
I had applied for Dropbox production status once more based on the changes I did with #1079. However, I still have not received positive feedback from the Dropbox team. The remaining part of Dropbox feedback from the first application that had not been addressed yet is #1090 (change the Dropbox access level to app folder access). I am now planning to fix that in order to have a successful Dropbox production status approval.

**Comment by TheArtForm** [comment]
I just found Ezra Bible and are blessed by a Linux/android cross platform study platform! Thank you for pusuing the Dropbox sync restoration. Know that your work is making a difference for the Kingdom!

**Reply by tobias-klein** [reply]
Thank you for your positive feedback @TheArtForm! That is encouraging!

The Dropbox issues hindering production status should all be closed in the next release 1.15.

Once that release is published users will connect to a newly set up App id. This then also means that the user limit will be temporarily reset. Then, after reaching 50 users on the new app id, I will re-apply for production status. I would estimate that this whole process (from the new app release to production status) could take anywhere between three and twelve months.

**Reply by TheArtForm** [reply]
Any chance you could just add nextcloud sync support?

**Reply by tobias-klein** [reply]
@TheArtForm Considering my limited capacity, I would say No here. As soon as you add multiple platforms, it multiplies the maintenance effort, too.

**Reply by TheArtForm** [reply]
I completely understand and respect that limitation. Thanks for what you have made. There is truly no other good option on Linux outside of a WINE solution, which is getting more difficult with the transition of my favorite DE to Wayland. It's also exciting to think of a sync between a Linux Bible and android. My only concern now is putting time and effort into tagging and studying without a functional backup solution (whether local or remote). It may hinder the application's full potential in my study and flow until backup is reliable. Thanks again for the excellent tool.

**Reply by tobias-klein** [reply]
> I completely understand and respect that limitation. Thanks for what you have made. There is truly no other good option on Linux outside of a WINE solution, which is getting more difficult with the transition of my favorite DE to Wayland. It's also exciting to think of a sync between a Linux Bible and android. My only concern now is putting time and effort into tagging and studying without a functional backup solution (whether local or remote). It may hinder the application's full potential in my study and flow until backup is reliable. Thanks again for the excellent tool.

Aside from the capabilities in the app itself, I would recommend you a personal backup solution.
On Linux, the database is stored in ~/.config/ezra-bible-app/ezra.sqlite
You can use any backup solution for backing up that file and then being sure that you have a backup archive available just in case. You can also inspect the content of that file with any sqlite database viewer if you are curious about the structure.

In the next version of Ezra Bible App there will be a built-in backup mechanism, which will be helpful in situations where the database got corrupted for one reason or another (see #1096). I have never had this situation myself so far. However, one user reported it recently. I sent him some tipps on how to recover based on the history available in Dropbox itself and based on that he successfully recovered.

**Reply by TheArtForm** [reply]
Well that will solve that problem. Excellent! Thank you so much for the insight and information.

**Comment by tobias-klein** [comment]
This is again relevant now and currently the ability to link additional users with their Dropbox account is frozen. When this happened last year I eventually reset the Dropbox app id in the next version of the app. I also had to make some changes in the Dropbox related implementation, based on requirements from the Dropbox folks. All changes necessary for Dropbox production support have been implemented and are working. I have now re-applied for Dropbox production mode. Let's see if the request goes through this time!

**Reply by tobias-klein** [reply]
They processed my request this week and declined it, claiming that installation fails on Windows and Ubuntu. I am still in touch with them via a ticket now and have provided a screencast of the Dropbox authentication flow to show that it is properly working. In parallel I need to check why installation could possibly fail on Windows and Ubuntu (no details provided).

**Reply by tobias-klein** [reply]
I'm happy to report that after following up with Dropbox support and providing the screen cast to them, they have now approved Ezra Bible App for production status! This means there is no user limit anymore to linking the App with your Dropbox account! :)
